File: src/leetcode/easy/961-n-repeated-element-in-size-2n-array.php
<?php

// https://leetcode.com/problems/n-repeated-element-in-size-2n-array/

// Runtime: 104 ms, faster than 18.18% of PHP online submissions for N-Repeated Element in Size 2N Array.

//class Solution {
//    function repeatedNTimes($A) {
//        $nums = [];
//        foreach ($A as $num) {
//            if (in_array($num, $nums)) {
//                return $num;
//            } else {
//                $nums[] = $num;
//            }
//        }
//        return false;
//    }
//}

// Runtime: 80 ms, faster than 90.91% of PHP online submissions for N-Repeated Element in Size 2N Array.

class Solution {
    function repeatedNTimes($A) {
        $len = count($A);
        for ($k = 1; $k <= 3; ++$k) {
            for ($i = 0; $i < $len - $k; $i++) {
                if ($A[$i] == $A[$i + $k]) {
                    return $A[$i];
                }
            }
        }
        return false;
    }
}

$A = [5,1,5,2,5,3,5,4];
echo (new Solution())->repeatedNTimes($A);

File: src/leetcode/easy/961-n-repeated-element-in-size-2n-array.go
package main

// https://leetcode.com/problems/n-repeated-element-in-size-2n-array/

import (
	"fmt"
)

// Runtime: 36 ms, faster than 97.67% of Go online submissions for N-Repeated Element in Size 2N Array.
// Memory Usage: 5.2 MB, less than 100.00% of Go online submissions for N-Repeated Element in Size 2N Array.

func repeatedNTimes(A []int) int {
	set := make(map[int]int)
	for _, v := range A {
		if _, ok := set[v]; ok {
			return v
		} else {
			set[v] = 1
		}
	}
	return 0
}

func main() {
	A := []int{1, 2, 3, 3}
	fmt.Println(A)
	fmt.Println(repeatedNTimes(A))
}
